### DISCUSSION TITLE
struggling with deployment

### DISCUSSION THREAD

**Discussion by fredbouin** [statamic/cms#4434] [Troubleshooting] [answered]
I am trying to deploy my website, I am hosting it with my bluehost account. I pulled the contents of my repo to my domain pointing directory.

I renamed .env.example to .env

I edited ```APP_URL``` in my .env file to reflect the domain name.
I ran ```php artisan key:generate``` which created an app key in ```APP_KEY```

I Installed php, I also ran ```composer install``` and refreshed stache

If I go to my domain on my browser I see: index of/ and the list of files and folders in the directory rather than a functioning copy of my website.

One tip I got on discord is that my server needs to be configured so /public is the root level. I don't understand how public can be my root level if it is a subdirectory. Is it something I need to configure somewhere?

**Comment by duncanmcclean** [comment]
You'll need to ask your hosting company (Bluehost) on how to do that. If you can't I'd suggest finding another hosting company.

**Reply by fredbouin** [reply]
ok I will follow up with them, otherwise from what I've mentioned does it look like I'm doing things right? Is there any specific things I should know in terms of configurations?

**Reply by duncanmcclean** [reply]
Yeah, I think you should be fine.

**Reply by jasonvarga** [reply]
It sounds like all you need to do it ask how you can make the `public` directory the webroot. Right now it sounds like the level above (the whole project itself) is the webroot.

**Reply by fredbouin** [reply]
Doesn't seem to work, thats fine I've been looking for a reason to ditch bluehost. I'll give the DO/Laravel Forge combo a try.

**Comment by zunnur-tiara** [comment, marked as answer]
Are you using Apache or Nginx with bluehost? If you are using apache my need to[ configure the .htaccess file](https://stackoverflow.com/questions/23635746/htaccess-redirect-from-site-root-to-public-folder-hiding-public-in-url). If you are using Nginx, you may follow https://github.com/statamic/cms/discussions/4434#discussioncomment-1453543 suggesting.

**Reply by fredbouin** [reply]
thanks @zunnur-tiara but I ended up going with what was recommended in docs, forge and digital ocean.

**Reply by zunnur-tiara** [reply]
That was much better. There is less of a bother.
